package mpsqual;

message GetPowerInfo {
	optional string name = 1; 
	optional string key = 2;
}

message SetPowerInfo {
    required string name = 1; 
    required string key = 2;
    required string value = 3;
}

message PowerInfo {
	enum Error {
		SUCCESS = 0;
		INVALID_MESSAGE = 1;
		INVALID_DEVICE_NAME = 2;
		INVALID_KEY = 3;
		I2C_MGR_CONN_ERROR = 4;
		GPIO_MGR_CONN_ERROR = 5;
		OPERATION_ERROR = 6;
	}
	
    required Error errorCode = 1; 
    repeated Value values = 2; 
}

message Value {
    required string name = 1;
    required string key = 2;
    required string value = 3;
}

